Marlin/src/lcd/extui/lib/mks_ui/tft_lvgl_configuration.cpp

```diff
@@ -434,7 +434,9 @@ void tft_lvgl_init() {
   disp_language_init();
   //spi_flash_read_test();
 
-  touch.init();
+  #if ENABLED(TOUCH_BUTTONS)
+    touch.init();
+  #endif
```

Comment:
MKS' UI doesn't use `TOUCH_BUTTONS`, so should this if statement be here?

Reply:
MKS use the same SPI for flash and touch. I did this if so you can use the marlin way (separated spi) or the original MKS way (undef TOUCH_BUTTONS -> spi flash/touch together).

I just fixed back the touch I messed up

Reply:
It could be dying for you because this touch code init code... now with the #if, it wont blow up.

Reply:
There are two different build environments for the Robin Nano, so this isn't needed when building with MKS' LVGL.

You'd enable TOUCH_BUTTONS when using `FSMC_GRAPHICAL_TFT` for 12864/Emulated mode:

https://github.com/MarlinFirmware/Marlin/blob/8474050adbe1bd4d363397cfedefaee996ac1e98/platformio.ini#L543-L546

You'd use  `TFT_LITTLE_VGL_UI` when building for the LVGL UI:

https://github.com/MarlinFirmware/Marlin/blob/8474050adbe1bd4d363397cfedefaee996ac1e98/platformio.ini#L630-L633

Reply:
Ok. In my case, I dont want the touch buttons on screen... The if is just to use the touch object (XPT2046 class).

After the last commit, black screen yet?

Reply:
The function: XPT2046_Rd_Addata only works if the board support have the touch and flash in the same SPI. MKS board is this way. Other boards dont. So I did this #if. 

So:
TFT_LITTLE_VGL_UI  with TOUCH_BUTTON  = separated SPI (my board)
TFT_LITTLE_VGL_UI without TOUCH_BUTTON  = same SPI (MKS board)
FSMC_GRAPHICAL_TFT with TOUCH_BUTTONS = any board that supports FSMC and touch